src/client/testing/page-objects/create-availability.ts

```diff
@@ -3,11 +3,103 @@ import RootPage from './root';
+  readonly btnSaveSession: Locator;
+  readonly sessionSuccessMsg: Locator;
 
   constructor(page: Page) {
     super(page);
     this.title = page.getByRole('heading', {
       name: 'Create availability',
     });
+
+    this.btnCreateAvailability = page.getByRole('button', {
+      name: 'Create availability',
+    });
+
+    this.sessionTittle = page.getByRole('heading', {
+      name: 'What type of session do you want to create?',
+    });
+
+    this.btnContinue = page.getByRole('button', {
+      name: 'Continue',
+    });
+
+    this.setAndCapacityTittle = page.getByRole('heading', {
+      name: 'Set time and capacity for your session',
+    });
+
+    this.btnSaveSession = page.getByRole('button', {
+      name: 'Save session',
+    });
+
+    this.sessionSuccessMsg = page.getByText(
+      'You have successfully created availability for the current site.',
+    );
+  }
+
+  async selectSession(sessionType: string) {
+    await this.page.getByRole('radio', { name: sessionType }).click();
+  }
+
+  async enterSingleDateSessionDate(day: string, month: string, year: string) {
+    const sessionDateFormGroup = this.page
+      .getByRole('group')
+      .filter({ has: this.page.getByText('Session date') });
+
+    await sessionDateFormGroup.getByLabel('Day').fill(day);
+    await sessionDateFormGroup.getByLabel('Month').fill(month);
+    await sessionDateFormGroup.getByLabel('Year').fill(year);
+  }
+
+  async enterWeeklySessionStartDate(day: string, month: string, year: string) {
+    const startDateFormGroup = this.page
+      .getByRole('group')
+      .filter({ has: this.page.getByText('Start date') });
+
+    await startDateFormGroup.getByLabel('Day').fill(day);
+    await startDateFormGroup.getByLabel('Month').fill(month);
+    await startDateFormGroup.getByLabel('Year').fill(year);
+  }
+
+  async enterWeeklySessionEndDate(day: string, month: string, year: string) {
+    const endDateFormGroup = this.page
+      .getByRole('group')
+      .filter({ has: this.page.getByText('End date') });
+
+    await endDateFormGroup.getByLabel('Day').fill(day);
+    await endDateFormGroup.getByLabel('Month').fill(month);
+    await endDateFormGroup.getByLabel('Year').fill(year);
+  }
+
+  async enterStartTime(hour: string, minute: string) {
+    await this.page.getByLabel('Session start time - hour').fill(hour);
+    await this.page.getByLabel('Session start time - minute').fill(minute);
+  }
+
+  async enterEndtTime(hour: string, minute: string) {
+    await this.page.getByLabel('Session end time - hour').fill(hour);
+    await this.page.getByLabel('Session end time - minute').fill(minute);
+  }
+
+  async noOfVaccinators(numberof: string) {
+    await this.page
+      .getByLabel('How many vaccinators or vaccination spaces do you have?')
+      .fill(numberof);
+  }
+
+  async appointmentLength(length: string) {
+    await this.page.getByLabel('How long are your appointments?').fill(length);
+  }
+
+  async addServices(serviceName: string) {
```

Comment:
This method name is plural, but only adds a singular service. It might be clearer if it was called just `addService`